test/test_charm.py

```diff
@@ -430,3 +431,49 @@ def test_containers(self):
         self.assertIsInstance(meta.containers['test2'], ContainerMeta)
         self.assertEqual(meta.containers['test1'].name, 'test1')
         self.assertEqual(meta.containers['test2'].name, 'test2')
+
+    def test_containers_storage(self):
+        meta = CharmMeta.from_yaml("""
+name: k8s-charm
+storage:
+  data:
+    type: filesystem
+    location: /test/storage
+  other:
+    type: filesystem
+    location: /test/other
+containers:
+  test1:
+    mounts:
+      - storage: data
+        location: /test/storage
+      - storage: other
+        location: /test/other
+""")
+        self.assertIsInstance(meta.containers['test1'], ContainerMeta)
+        self.assertIsInstance(meta.containers['test1'].mounts["data"], ContainerStorageMeta)
+        self.assertEqual(meta.containers['test1'].mounts["data"].location, '/test/storage')
+        self.assertEqual(meta.containers['test1'].mounts["other"].location, '/test/other')
+
+    def test_containers_storage_multiple_mounts(self):
+        meta = CharmMeta.from_yaml("""
+name: k8s-charm
+storage:
+  data:
+    type: filesystem
+    location: /test/storage
+containers:
+  test1:
+    mounts:
+      - storage: data
+        location: /test/storage
+      - storage: data
+        location: /test/other
```

Comment:
not sure what this non-uniqueness means.
does it mean that the same data found in the charm container under `/test/storage` is mounted twice in the workload container?

Reply:
Yes. Specifically, see "Using subPath" in the kubernetes documentation.

The reverse is also possible with projected volumes (primarily for secrets), but I don't think Juju has gotten to that point yet.